components/server/src/workspace/workspace-service.ts

```diff
@@ -813,6 +819,53 @@ export class WorkspaceService {
             }
         }
     }
+
+    public async controlAdmission(
+        userId: string,
+        workspaceId: string,
+        level: "owner" | "everyone",
+        check: (workspace: Workspace, instance?: WorkspaceInstance) => Promise<void> = async () => {},
+    ): Promise<void> {
+        await this.auth.checkPermissionOnWorkspace(userId, "access", workspaceId);
```

Comment:
We should probably have a specific permission for changing the sharing properties in another PR. At least if we decide to keep this feature.

Reply:
Jep. Was on the brink to introduce one, then kept the current behavior. :+1: